nilearn/mass_univariate/permuted_least_squares.py

```diff
@@ -687,12 +688,25 @@ def permuted_ols(
 
     n_samples, n_regressors = tested_vars.shape
 
-    # check if explanatory variates is intercept (constant) or not
+    # check if explanatory variates contain an intercept (constant) or not
     if n_regressors == np.unique(tested_vars).size == 1:
         intercept_test = True
     else:
         intercept_test = False
 
+    # check if confounding vars contains an intercept
+    if confounding_vars is not None:
+        constants = []
+        for column in range(confounding_vars.shape[1]):
+            if np.unique(confounding_vars[:, column]).size == 1:
+                constants.append(column)
+        if (intercept_test and len(constants) == 1) or len(constants) > 1:
+            raise ValueError('More than one intercepts '
```

Comment:
In that  case, should we raise en error ? Why not rather spit a warning and remove an intercept ?

Reply:
Perhaps even better, the 3rd line should be `if not model_intercept and not intercept_test:`